### PR TITLE
[mlir] add overload `createDIScopeForLLVMFuncOp` function

### PR DESCRIPTION
**Pull request by Observer007**
follow up work of https://github.com/llvm/llvm-project/pull/106229, add create pass overload function to create pass.

### CODE REVIEW

File: mlir/include/mlir/Dialect/LLVMIR/Transforms/Passes.h [hunk 1]
```diff
@@ -23,6 +23,10 @@ namespace LLVM {
 /// Create a pass to add DIScope to LLVMFuncOp that are missing it.
 std::unique_ptr<Pass> createDIScopeForLLVMFuncOpPass();
 
+struct DIScopeForLLVMFuncOpOptions;
+std::unique_ptr<Pass>
+createDIScopeForLLVMFuncOpPass(const DIScopeForLLVMFuncOpOptions &options);
+
```
Comment (joker-eph) [inline mlir/include/mlir/Dialect/LLVMIR/Transforms/Passes.h:29]: I think that removing the `let constructor = ` from the TableGen declaration for this pass will see the createXXX method auto-generated. Can you try this?

File: mlir/lib/Dialect/LLVMIR/Transforms/DIScopeForLLVMFuncOp.cpp [hunk 2]
```diff
@@ -86,6 +86,10 @@ namespace {
 /// Add a debug info scope to LLVMFuncOp that are missing it.
 struct DIScopeForLLVMFuncOp
     : public LLVM::impl::DIScopeForLLVMFuncOpBase<DIScopeForLLVMFuncOp> {
+  DIScopeForLLVMFuncOp() = default;
+  DIScopeForLLVMFuncOp(const mlir::LLVM::DIScopeForLLVMFuncOpOptions &options)
```
Comment (Dinistro) [inline mlir/lib/Dialect/LLVMIR/Transforms/DIScopeForLLVMFuncOp.cpp:90]:
```suggestion
  explicit DIScopeForLLVMFuncOp(const mlir::LLVM::DIScopeForLLVMFuncOpOptions &options)
```

Except you want implicit conversion, ofc.

**Review by Dinistro — COMMENTED** [review]
Small drive by comment

File: mlir/include/mlir/Dialect/LLVMIR/Transforms/Passes.h (hunk 1 above)
Comment (Observer007) [reply mlir/include/mlir/Dialect/LLVMIR/Transforms/Passes.h:29]: Thanks, then the changes become more concise.

File: mlir/lib/Dialect/LLVMIR/Transforms/DIScopeForLLVMFuncOp.cpp (hunk 2 above)
Comment (Observer007) [reply mlir/lib/Dialect/LLVMIR/Transforms/DIScopeForLLVMFuncOp.cpp:90]: Thanks, using `using Base::Base` instead.

**Review by Dinistro — APPROVED** [review]
If this generates all the pass creation function you require, this is LGTM!
Please make sure that you update the commit message accordingly, as it no longer matches what this change does.

**Review by joker-eph — APPROVED** [review]
LG, thanks!

**Pull request merged**